libraries/AP_RCTelemetry/AP_CRSF_Telem.h

```diff
@@ -307,16 +308,17 @@ class AP_CRSF_Telem : public AP_RCTelemetry {
     bool _get_telem_data(AP_RCProtocol_CRSF::Frame* data);
     bool _process_frame(AP_RCProtocol_CRSF::FrameType frame_type, void* data);
 
-    TelemetryPayload _telem;
-    uint8_t _telem_size;
-    uint8_t _telem_type;
-    AP_RCProtocol_CRSF::RFMode _telem_rf_mode;
-    // reporting telemetry rate
-    uint32_t _telem_last_report_ms;
-    uint16_t _telem_last_avg_rate;
-
-    bool _telem_pending;
-    bool _enable_telemetry;
+    struct {
```

Comment:
This structural change has made the PR much, much larger than it needed to be. I would prefer to see a NFC PR for the refactoring and then a different PR for the ELRS change

Reply:
I'll put the refactoring aside for now